samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Views/Manage/SelectAuthenticator.cshtml

```diff
@@ -12,16 +14,32 @@
             @using (Html.BeginForm("SelectAuthenticatorAsync", "Manage", new { ReturnUrl = ViewBag.ReturnUrl }, FormMethod.Post, new { @class = "form-horizontal", role = "form" }))
             {
                 @Html.AntiForgeryToken()
-                if (Model.CanSkip)
-                 {
-                     <h4>Set up optional authenticators</h4>
-                 }
-                 else
+
+                if (isChallengeFlow)
+                {
+                    <h4>Verify it's you with an authenticator</h4>
+                }
+                else
                 {
+
+                    if (Model.CanSkip)
+                    {
+                     <h4>Set up optional authenticators</h4>
+                    }
+                    else
+                    {
                      <h4>Set up required</h4>
+                    }
+                }
 
+                if (isChallengeFlow)
```

Comment:
There is a lot of "logic" in this template.  I realize it is just if/else logic but generally speaking templates should have as little logic as possible.  I think this would be easier to read if the "variables" were encapsulated in the SelectAuthenticatorViewModel; if the logic were moved to the viewModel your markup could look something like:

```
@Model.Title
@Model.SubTitle
```

The logic is only determining what `h4` and `h5` to display.

Reply:
We'll need to address this later on during the sample refactoring